src/core/thread/address_resolver.cpp

```diff
@@ -291,7 +291,7 @@ otError AddressResolver::Resolve(const Ip6::Address &aEid, uint16_t &aRloc16)
         else if (entry->mTimeout == 0 && entry->mRetryTimeout == 0)
         {
             entry->mTimeout = kAddressQueryTimeout;
-            SendAddressQuery(aEid);
+            SuccessOrExit(error = SendAddressQuery(aEid));
```

Comment:
Move this one line up to avoid setting the timeout if `SendAddressQuery()` returns error?

Reply:
I also moved the other verification.